lib/rust/ensogl/component/grid-view/src/basic.rs

```diff
@@ -0,0 +1,182 @@
+//! A module defining the [`BasicGridView`] with all helper structures.
+
+use crate::prelude::*;
+use ensogl_core::display::shape::*;
+
+use crate::scrollable;
+use crate::EntryFrp;
+
+use ensogl_core::application::command::FrpNetworkProvider;
+use ensogl_core::application::frp::API;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_text as text;
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+/// The background of single Entry. The actually displayed rectangle is shrunk by [`PADDING_PX`]
+/// from the shape size, to avoid antialiasing glitches.  
+pub mod entry_background {
+    use super::*;
+
+    /// A padding added to the background rectangle to avoid antialiasing glitches.
+    pub const PADDING_PX: f32 = 5.0;
+
+    ensogl_core::define_shape_system! {
+        (style:Style, color: Vector4) {
+            let shape_width  : Var<Pixels> = "input_size.x".into();
+            let shape_height : Var<Pixels> = "input_size.y".into();
+            let width = shape_width - 2.0.px() * PADDING_PX;
+            let height = shape_height - 2.0.px() * PADDING_PX;
+            Rect((width, height)).fill(color).into()
+        }
+    }
+}
+
+
+
+// ===================
+// === EntryParams ===
+// ===================
+
+/// The parameters of [`BasicGridView`]`s entries.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryParams {
+    pub bg_color:    color::Rgba,
+    pub bg_margin:   f32,
+    pub font:        ImString,
+    pub text_offset: f32,
+    pub text_size:   text::Size,
+    pub text_color:  color::Rgba,
+}
+
+impl Default for EntryParams {
+    fn default() -> Self {
+        Self {
+            bg_color:    color::Rgba::transparent(),
+            bg_margin:   0.0,
+            font:        text::typeface::font::DEFAULT_FONT.into(),
+            text_offset: 7.0,
+            text_size:   text::Size(14.0),
+            text_color:  default(),
+        }
+    }
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+// === EntryData ===
+
+/// An internal structure of [`Entry`], which may be passed to FRP network without risk of memory
+/// leaks.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryData {
+    display_object: display::object::Instance,
+    pub label:      text::Area,
+    pub background: entry_background::View,
+}
+
+impl EntryData {
+    fn new(app: &Application, text_layer: &Option<Layer>) -> Self {
+        let logger = Logger::new("list_view::entry::Label");
```

Comment:
Use the new logging that is exposed in prelude instead of our old logger, please.

Reply:
I have to pass something to display::object::Instance. Can I do it in the new logging somehow?